cmd/charm/charmcmd/logout.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the GPLv3, see LICENCE file for details.
+
+package charmcmd
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/juju/cmd"
+	"github.com/juju/persistent-cookiejar"
+	"golang.org/x/net/publicsuffix"
+	"gopkg.in/errgo.v1"
+	"gopkg.in/macaroon.v1"
+)
+
+type logoutCommand struct {
+	cmd.CommandBase
+}
+
+var logoutDoc = `
+The logout command removes all security credentials for charm store.
+
+   charm logout
+`
+
+func (c *logoutCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "logout",
+		Purpose: "logout from charm store",
+		Doc:     logoutDoc,
+	}
+}
+
+func (c *logoutCommand) Run(ctxt *cmd.Context) error {
+	// Delete any Ubuntu SSO token.
+	if err := os.Remove(ussoTokenPath()); err != nil && !os.IsNotExist(err) {
+		return errgo.New("cannot remove Ubuntu SSO token")
+	}
+
+	// Delete any login cookies.
+	jar, err := cookiejar.New(&cookiejar.Options{
```

Comment:
Alternatively:

	client, err := newCharmStoreClient(ctxt, c.username, c.password)
	if err != nil {
		return errgo.Notef(err, "cannot create the charm store client")
	}
	defer client.jar.Save()
        u, err := url.Parse(client.ServerURL())
        ...

which avoids duplicating the cookiejar New code.